stubs/Flask-SQLAlchemy/flask_sqlalchemy/model.pyi

```diff
@@ -22,6 +20,6 @@ class BindMetaMixin(type):
 
 class DefaultMeta(NameMetaMixin, BindMetaMixin, DeclarativeMeta): ...
 
-class Model(Generic[_ModelT]):
-    query_class: type[Query[_ModelT]] | None
-    query: Query[_ModelT] | None
+class Model:
+    query_class: type[Query[Self]] | None
+    query: Query[Self] | None
```

Comment:
We can't use `typing.Self` in typeshed yet, as mypy doesn't support it currently. We _can_ use `_typeshed.Self`, however (which is just a normal `TypeVar`). Because (unlike `typing.Self`) it's just a `TypeVar`, however, we need to be slightly hacky when we're using it. You'd do it like this:

```suggestion
    # These aren't actually properties at runtime, but we treat them as such so we can use `self` types
    # without using typing.Self
    @property
    def query_class(self: Self) -> type[Query[Self]] | None: ...
    @query_class.setter
    def query_class(self: Self, value: type[Query[Self]] | None) -> None: ...
    @property
    def query(self: Self) -> Query[Self] | None: ...
    @query.setter
    def query(self: Self, value: Query[Self] | None) -> None: ...
```

Reply:
Note: I haven't verified whether this change is correct or not yet; this is just how you'd do it, providing it _is_ correct :)

Reply:
Can we add ifs to check python version by >= 3.11?

Reply:
No, as mypy currently does not support it on any version.

Pyright supports `typing.Self` on 3.11+, and `typing_extensions.Self` on all Python versions, so if it were just a matter of the Python version on which the feature is available, we could just use `typing_extensions.Self`. But that's not the issue: the issue is that mypy hasn't yet added support for this new feature at all, on any Python version.

Reply:
@AlexWaygood can I open new PR to avoid committing spam? (sorry for that)

Reply:
It'll all be squashed into a single commit before merging, don't worry about making a mess!